tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,724 @@
+            missingRequiredValue = true;
+        }
+        if (descriptor.getFileSystemRealmName() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockFilesystemRealmName(count));
+            missingRequiredValue = true;
+        }
+        if(descriptor.getHashEncoding() == null) {
+            descriptor.setHashEncoding(Encoding.BASE64);
+        }
+        if(descriptor.getEncoded() == null) {
+            descriptor.setEncoded(true);
+        }
+        if(descriptor.getLevels() == 0) {
+            descriptor.setLevels(2);
+        }
+        if(descriptor.getCreateCredentialStore() == null) {
+            descriptor.setCreateCredentialStore(false);
+        }
+        if (missingRequiredValue) {
+            descriptor.reset();
+        }
+    }
+
+    /**
+     * Handles creating the Elytron filesystem-realm from the descriptor array
+     *
+     * @throws Exception Exception to be handled by Elytron Tool
+     */
+    private void createFileSystemRealm() throws Exception {
+        for (Descriptor descriptor : descriptors) {
+            System.out.println("Creating encrypted realm for: " + descriptor.getInputRealmLocation());
+            if (descriptor.getFileSystemRealmName() == null || descriptor.getInputRealmLocation() == null || descriptor.getOutputRealmLocation() == null) {
+                continue;
+            }
+            CredentialStore credentialStore;
+            if (descriptor.getCreateCredentialStore()) {
+                descriptor.setCredentialStore(Paths.get(descriptor.getOutputRealmLocation(), "mycredstore.cs").toString());
+                descriptor.setSecretKeyAlias("key");
+            }
+            // check if credential-store and secret-key-alias are both specified, or both null
+            if (! (descriptor.getCredentialStore() == null ^ descriptor.getSecretKeyAlias() == null) ){
+                String csType = PropertiesCredentialStore.NAME;
+                try {
+                    credentialStore = CredentialStore.getInstance(csType);
+                } catch (NoSuchAlgorithmException e) {
+                    // fallback to load all possible providers
+                    credentialStore = CredentialStore.getInstance(csType, getProvidersSupplier(null));
+                }
+                Map<String, String> implProps = new HashMap<>();
+                implProps.put("create", String.valueOf(descriptor.getCreateCredentialStore()));
+                implProps.put("location", descriptor.getCredentialStore());
+                implProps.put("modifiable", Boolean.TRUE.toString());
+                credentialStore.initialize(implProps);
+                try {
+                    credentialStore.retrieve(descriptor.getSecretKeyAlias(), SecretKeyCredential.class).getSecretKey();
+                    System.out.println("Found credential store and alias, using pre-existing key");
```

Comment:
Should this be here?

Reply:
If we do use prints like this in other tool commands, we should at least get the message into ElytronToolMessages.